src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -269,6 +372,15 @@ impl<'tcx> LoanPath<'tcx> {
         LoanPath { kind: kind, ty: ty }
     }
 
+    pub fn ref_node_id<'a>(&self) -> ast::NodeId {
```

Comment:
I wrote this well over 6 months ago, so during that time someone might have written a helper for this. Back then I only saw other usages of this in `match` statements. The purpose here is to get the `NodeId` for the value that a `Loan` is being created for. I should have come back to this because I never really learned what Upvar, Downcast and Extend actually do.